src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CSharp.Analyzers/ApiControllerAnalyzer.cs

```diff
@@ -0,0 +1,108 @@
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterSyntaxNodeAction(AnalyzeNode, cs.SyntaxKind.SimpleBaseType);
+            context.RegisterSyntaxNodeAction(AnalyzeNode, vb.SyntaxKind.InheritsStatement);
+        }
+
+        private void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            var namedTypeSymbol = context.ContainingSymbol as INamedTypeSymbol;
+
+            if (namedTypeSymbol != null && namedTypeSymbol.BaseType != null
+                && namedTypeSymbol.BaseType.ToDisplayString().Equals($"{BadNamespace}.{BadClassName}", StringComparison.Ordinal))
+            {
+                if (context.Compilation.Language == LanguageNames.CSharp)
+                {
+                    ReportCSharpSyntax(context);
+                }
+                else if (context.Compilation.Language == LanguageNames.VisualBasic)
+                {
+                    ReportVisualBasicSyntax(context);
+                }
+            }
+        }
+
+        private static void ReportCSharpSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var diagnostic = Diagnostic.Create(Rule, context.Node.GetLocation(), context.Node.ToString());
+            context.ReportDiagnostic(diagnostic);
+        }
+
+        /// <summary>
+        /// Uses syntax analysis to highlight the correct part of the node. At this point
+        /// the context.Node looks like 'Inherits System.Web.Http.ApiController' and we
+        /// do not want to highlight the VisualBasic.SyntaxKind.InheritsKeyword
+        /// </summary>
+        /// <param name="context"></param>
+        private static void ReportVisualBasicSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var baseClass = context.Node.DescendantNodes()
+                .OfType<vb.Syntax.QualifiedNameSyntax>()
+                .FirstOrDefault() as SyntaxNode;
+
+            if (baseClass is null)
+            {
+                baseClass = context.Node.DescendantNodes()
+                .OfType<vb.Syntax.IdentifierNameSyntax>()
+                .FirstOrDefault();
+            }
+
+            var diagnostic = Diagnostic.Create(Rule, baseClass.GetLocation(), baseClass.ToString());
+            context.ReportDiagnostic(diagnostic);
+        }
+
+        public static Project AddMetadataReferences(Project project)
```

Comment:
This isn't the right place for this method to live. The source updater shouldn't have a dependency on any particular extension.

Reply:
I think the extension needs a way to declare a requirement of what it needs to run. Do you have another recommendation about how to declare this requirement?

Also, I liked that the requirement of the Analyzer is declared with the Analyzer but I can see that it could be more clearly highlighted.